src/test/java/org/openrewrite/staticanalysis/RemovePrivateFieldUnderscoresTest.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2025 the original author or authors.
+ * <p>
+ * Licensed under the Moderne Source Available License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://docs.moderne.io/licensing/moderne-source-available-license
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.staticanalysis;
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.DocumentExample;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+class RemovePrivateFieldUnderscoresTest implements RewriteTest {
+
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec.recipe(new RemovePrivateFieldUnderscores());
+    }
+
+    @DocumentExample
+    @Test
+    void removesPrefixUnderscore() {
+        rewriteRun(
+          //language=java
+          java(
+            """
+              public class ParseLocation {
+                  private String _ruleName;
```

Comment:
I think it would be good to add a test showing either `class_` or `_class` to show we do not end up renaming to restricted keywords.